snap/snapcraft.yaml

```diff
@@ -0,0 +1,59 @@
+name: pebble
+adopt-info: pebble-release-data
+summary: Pebble - take control of your internal daemons!
+description: |
+  Pebble helps you to orchestrate a set of local service processes
+  as an organized set. It resembles well-known tools such as supervisord,
+  runit, or s6, in that it can easily manage non-system processes
+  independently from the system services, but it was designed with unique
+  features that help with more specific use cases.
+
+  **Usage**
+
+  * The Pebble snap's default directory is
+  `$HOME/snap/pebble/<rev>/var/lib/pebble/default`. This directory is **not**
+  created at installation time.
+
+  **Documentation**
+
+  To learn more about Pebble please check the project's documentation at
+  https://github.com/canonical/pebble.
+issues: https://github.com/canonical/pebble/issues
+source-code: https://github.com/canonical/pebble
+license: GPL-3.0
+base: bare
+build-base: core22
+confinement: classic
+
+apps:
+  pebble:
+    command: pebble-wrapper
+
+parts:
+  pebble:
+    plugin: go
+    source: .
+    build-snaps:
+      - go
+    build-environment:
+      - CGO_ENABLED: 0
+      - GOFLAGS: -ldflags=-w -ldflags=-s
```

Comment:
FWIW, we use `-trimpath -ldflags='-s -w'` in the GitHub Actions that builds the official binaries, so probably best to replicate that here.

I'm curious, how does this know to build the `./cmd/pebble` package?

Reply:
Alternatively I suppose we could download one of the official binaries, now that those are being built?

Reply:
Good catch. Thanks, I'll add `-trimpath` to the GOFLAGS (see https://github.com/canonical/pebble/pull/179/commits/c49bb04a426edf10009b5b62c4ea4f0521a4c177). The `go` plugin installs for `./...` ([ref](https://github.com/snapcore/snapcraft/blob/0594429b75bfc663a59f583e42639239c1568e0a/snapcraft_legacy/plugins/v2/go.py#L79)).

Wrt to taking the binaries instead of re-compiling...I'd need to think about it. I'd say it's not possible atm the moment, mainly for 2 reasons:
 1. the snap is built by the store, automatically. We can't customize (afaik) the triggering conditions for that store-side build, which means we can't guarantee the Pebble binary has already been built whenever the store build takes place
 2. the pebble binaries are only released to a predictable place `on: release`, otherwise they are only uploaded as a build artifact of the GH action, which means: 1) it would be cumbersome for snapcraft.yaml to infer where to get that binary from, and 2) those binaries will only be there for a limited amount of time (the default retention time is 90 days)

Reply:
Yep, that's fair -- thanks